docs/src/components/ButtonLink.tsx

```diff
@@ -17,60 +23,137 @@ type Props<T extends React.ElementType = typeof DefaultComponent> = PolymorphicP
   T
 >;
 
-const StyledComponent = styled(DefaultComponent)<Props>`
-  ${({ theme, type, size }) => css`
+function getColor({ theme, type }: { theme: ThemeShape } & Pick<OwnProps, "type">): string {
+  return type === "secondary" ? theme.orbit.paletteProductDark : theme.orbit.paletteWhite;
+}
+
+function getBgColor({
+  theme,
+  type,
+  color,
+}: { theme: ThemeShape } & Pick<OwnProps, "type" | "color">): string {
+  switch (type) {
+    case "primary":
+      return theme.orbit.backgroundButtonPrimary;
+    case "secondary":
+      return theme.orbit.paletteProductLight;
+    default:
+      return typeof color !== "undefined" ? color : "";
+  }
+}
+
+function getBgColorHover({
+  theme,
+  type,
+  color,
+  dark,
+}: { theme: ThemeShape } & Pick<OwnProps, "type" | "color" | "dark">): string {
+  if (dark) {
+    const [h, s, l] = hex2hsl(getBgColor({ theme, type, color })).map(num => Math.round(num));
+    return `hsl(${h}, ${s}%, ${dark ? l + 4 : l - 4}%)`;
+  }
+
+  switch (type) {
+    case "primary":
+      return theme.orbit.backgroundButtonPrimaryHover;
+    case "secondary":
+      return theme.orbit.paletteProductLightHover;
+    default:
+      return "";
+  }
+}
+
+function getBgColorFocus({
+  theme,
+  type,
+  color,
+  dark,
+}: { theme: ThemeShape } & Pick<OwnProps, "type" | "color" | "dark">): string {
+  if (typeof color !== "undefined" || dark) {
+    return getBgColorHover({ theme, type, color, dark });
+  }
+
+  switch (type) {
+    case "primary":
+      return theme.orbit.paletteProductNormal;
+    case "secondary":
+      return theme.orbit.paletteProductLightHover;
+    default:
+      return "";
+  }
+}
+
+function getBgColorActive({
+  theme,
+  type,
+  color,
+  dark,
+}: { theme: ThemeShape } & Pick<OwnProps, "type" | "color" | "dark">): string {
+  if (typeof color !== "undefined" || dark) {
+    const [h, s, l] = hex2hsl(getBgColor({ theme, type, color })).map(num => Math.round(num));
```

Comment:
i think you can have it like that:

```suggestion
    const [h, s, l] = hex2hsl(getBgColor({ theme, type, color })).map(Math.round);
```